### ISSUE TITLE
Webcam OSX Sierra

### ISSUE THREAD

**Issue by ricardolmsilva** [qTox/qTox#4866]
OS X Sierra 10.12.65
qTox version: v1.13.0
Commit hash:  a66af1b128f7293bc08718ce2cc064f
toxcore:  0.1.0
Qt:  5.9.3

Hi, im using qTox v1.13.0 on OSX Sierra and the Webcam in qTox is not detect as you can see in the screenshot below.

![33979210-3c79ae7c-e09b-11e7-98e2-7db73e4093ed](https://user-images.githubusercontent.com/34528668/34066690-9bab0812-e20a-11e7-9068-17b9f9b7517f.png)

**Comment by sudden6** [comment]
Please attach the log output when this happens.

**Comment by ricardolmsilva** [comment]
Where can i get this file?
Its this?

[log.txt](https://github.com/qTox/qTox/files/1565063/log.txt)

**Comment by sudden6** [comment]
Yes, this was the correct file.

Relevant part:
```
[01:15:48.023 UTC] widget/form/settings/avform.cpp:345 : Debug: available Modes:
[01:15:48.024 UTC] widget/form/settings/avform.cpp:272 : Debug: selected Modes:
[01:15:48.024 UTC] widget/form/settings/avform.cpp:281 : Debug: width: 1280, height: 720, FPS: 30.000000, pixel format: unknown

[01:15:48.024 UTC] widget/form/settings/avform.cpp:281 : Debug: width: 640, height: 480, FPS: 30.000000, pixel format: unknown

[01:15:48.024 UTC] widget/form/settings/avform.cpp:281 : Debug: width: 320, height: 240, FPS: 30.000000, pixel format: unknown
```

**Comment by sudden6** [comment]
The symptoms look the same as in #4764

especially the `pixel format: unknown` stuff.

**Comment by ricardolmsilva** [comment]
Yes, I've seen this issue before but have not seen a solution for me.
In addition said my system is osx and not windows.
I do not know how to solve this problem :(

**Comment by fidi456** [comment]
Hello,
I have the same problem with High Sierra. Log is almost the same. Have tested with a cheap USB Webcam this works. But if I disconnect the webcam and do not refresh devices, go from USB to facetime and back to USB, Facetime works until refresh Devices or close the program. FaceTime worked in version 1.10, but when you stopped a video call in this version, the software crashed completely :(

**Comment by sudden6** [comment]
@fidi456 can you provide the logs of it working and not working?

We'll have to try to figure this out only with the logs :/

**Comment by fidi456** [comment]
Hello, here my log 0x14100000046d0824 is USB Cam and 0x1a11000005ac8509 FaceTime ID. Has worked again, USB Cam then Facetime, USB disconnect back to USB and Facetime worked.
[LOG.txt](https://github.com/qTox/qTox/files/1568265/LOG.txt)

**Comment by ricardolmsilva** [comment]
I dont know if it help to solve the problem, but in "utox" webcam works! 👍

**Comment by sudden6** [comment]
Can any of you run a git bisect from the last working version to the first broken release to find the offending commit?

**Comment by ricardolmsilva** [comment]
sorry, my english is a little bad, and i dont understanded what u need for real, can u explain an easy way to do it?

**Comment by sudden6** [comment]
Well, this could require some effort from your side if you are not familiar with `git` or compiling from source, but here are the steps:

- install `git`
- use `git clone https://github.com/qTox/qTox` to download the source code
- setup the build environment for qTox, see https://github.com/qTox/qTox/blob/master/INSTALL.md#os-x
- use `git bisect start` to start the process
- use `git bisect good v1.11.0` to mark the last known good version
- use `git bisect bad v1.13.0` to mark the first known bad version
- use `git bisect` to checkout some commit, rebuild qTox and test if it detects your webcam

**Comment by fidi456** [comment]
Hello, currently have no time, at the weekend I will compile the current git version and the version where Facetime works and create the log. For information I have tried the git version ever and the error was the same, unfortunately log deleted

**Comment by sudden6** [comment]
@fidi456 that would be great if you could try the above process to find the breaking commit.

**Comment by Hintsmth** [comment]
@ricardolmsilva we haven't still seen a solution for our windows-problem either.

**Comment by Diadlo** [comment]
@Hintsmth If it's the same issue, `bisect` can help to find the root of problem

**Comment by fidi456** [comment]
Hello, I had a little time here now the logs, compile with git bisect and error
[bitsectbuild.txt](https://github.com/qTox/qTox/files/1579441/bitsectbuild.txt)
bisect reset an compile 1.13 with facetime Problem, log shell start
[shellStart1.13.txt](https://github.com/qTox/qTox/files/1579445/shellStart1.13.txt)
log 1.11  start with facetime works only after ending a video call, the software crashes completely
[Log1.11.txt](https://github.com/qTox/qTox/files/1579450/Log1.11.txt)
Crash 
[Crash.txt](https://github.com/qTox/qTox/files/1579453/Crash.txt)
I hope that helps finding the bug

**Comment by sudden6** [comment]
@fidi456 If a commit doesn't build for whatever reason use `git bisect skip` to skip that particular commit.

**Comment by sudden6** [comment]
@fidi456 If a commit doesn't build for whatever reason use `git bisect skip` to skip that particular commit.

EDIT: I have a theory about the error now:

- Log1.11.txt shows the FPS in the video device as float with decimal dot.
- shellStart1.13 show the following interesting part
```
...
[13:54:25.550 UTC] widget/form/settings/avform.cpp:301 : Debug: width: 160, height: 120, FPS: 29.969999, pixel format: unknown

[13:54:25.550 UTC] video/camerasource.cpp:383 : Debug: Closing device  "none"
[13:54:25.550 UTC] video/camerasource.cpp:267 : Debug: Opening device  "0xfa20000005ac850b"
[avfoundation @ 0x7fed6e02c800] Selected framerate (29,000000) is not supported by the device
[avfoundation @ 0x7fed6e02c800] Supported modes:
[avfoundation @ 0x7fed6e02c800]   160x120@[29,970000 29,970000]fps
[avfoundation @ 0x7fed6e02c800]   160x120@[25,000000 25,000000]fps
[avfoundation @ 0x7fed6e02c800]   160x120@[23,999981 23,999981]fps
[avfoundation @ 0x7fed6e02c800]   160x120@[14,999993 14,999993]fps
[avfoundation @ 0x7fed6e02c800]   176x144@[29,970000 29,970000]fps
[avfoundation @ 0x7fed6e02c800]   176x144@[25,000000 25,000000]fps
[avfoundation @ 0x7fed6e02c800]   176x144@[23,999981 23,999981]fps
[avfoundation @ 0x7fed6e02c800]   176x144@[14,999993 14,999993]fps
[avfoundation @ 0x7fed6e02c800]   320x240@[29,970000 29,970000]fps
...
```
Where avfoundation complains about the FPS not being found, because it misses the comma part.

EDIT2: seems like the qtox log doesn't capture the errors from avfoundation and they are only visible in the terminal. Please everybody in this thread only use the terminal output log for this issue.

**Comment by sudden6** [comment]
The above pull request could fix the issue for Mac OS and Windows. Please test and report back :)

**Comment by fidi456** [comment]
@sudden6 , that with the comma and point was already good. Only one problem has to be considered, start in the shell then this error comes.
[shellstart1.13.txt](https://github.com/qTox/qTox/files/1581800/shellstart1.13.txt)
But if you start directly from the icon in the Applications folder without shell, it works. The crash when the video call ends I test later
![screen](https://user-images.githubusercontent.com/34652984/34290080-dc1cc0b2-e6f5-11e7-895b-9f87894b80b8.jpg)

**Comment by fidi456** [comment]
@sudden6 completely forgotten, here is the log of the working facetime, started directly in the Applications folder
[qtoxlog.txt](https://github.com/qTox/qTox/files/1581874/qtoxlog.txt)

**Comment by fidi456** [comment]
Answer and end a video call works without a crash, here's the log.
A few small bugs could still be. If I notice something else I post it. Thank you
[qtoxworks.txt](https://github.com/qTox/qTox/files/1581996/qtoxworks.txt)

**Comment by sudden6** [comment]
@fidi456 so the Facetime camera works now with the patches in the PR?

The bug with qTox crashing after ending a video call should be fixed in > 1.13, for other bugs don't hesitate to open a new issue.

**Comment by jpp4** [comment]
My bug report
[Debog.log](https://github.com/qTox/qTox/files/3061788/Debog.log)
